common-fragment-V1/src/main/java/org/pentaho/hbase/shim/common/CommonHBaseConnection.java

```diff
@@ -121,42 +123,46 @@ public static Integer getMajorVersionNumber( String shimConfigurationId ) {
   }
 
   @Override
-  public void configureConnection( Properties connProps, List<String> logMessages ) throws Exception {
+  public void configureConnection( Properties connProps, NamedCluster namedCluster,
+                                   List<String> logMessages ) throws Exception {
     ClassLoader cl = Thread.currentThread().getContextClassLoader();
     Thread.currentThread().setContextClassLoader( getClass().getClassLoader() );
     try {
       String defaultConfig = connProps.getProperty( DEFAULTS_KEY );
       String siteConfig = connProps.getProperty( SITE_KEY );
       String zookeeperQuorum = connProps.getProperty( ZOOKEEPER_QUORUM_KEY );
       String zookeeperPort = connProps.getProperty( ZOOKEEPER_PORT_KEY );
-      String namedCluster = connProps.getProperty( NAMED_CLUSTER );
       boolean namedClusterIsMapr = Boolean.parseBoolean( connProps.getProperty( SHIM_IS_MAPR, "false" ) );
 
       m_config = new Configuration();
+      BiConsumer<InputStream, String > consumer = ( is, filename ) ->
+      {
+        m_config.addResource( is, filename );
+      };
       try {
         if ( namedClusterIsMapr ) {
-          ShimConfigsLoader.addConfigsAsResources( namedCluster, m_config::addResource,
+          ShimConfigsLoader.addConfigsAsResources( namedCluster, consumer,
```

Comment:
Can this method be made a member of the NamedCluster object so we can get rid of ShimConfigsLoader?